src/planner/binder/expression/bind_comparison_expression.cpp

```diff
@@ -125,6 +125,14 @@ BindResult ExpressionBinder::BindExpression(ComparisonExpression &expr, idx_t de
 	// add casts (if necessary)
 	left.expr = BoundCastExpression::AddCastToType(move(left.expr), input_type);
 	right.expr = BoundCastExpression::AddCastToType(move(right.expr), input_type);
+	if (where_clause) {
+		if (left.expr->type == ExpressionType::OPERATOR_CAST && left.expr->return_type.id() == LogicalTypeId::ENUM) {
```

Comment:
Is there a reason to only do this in the `WHERE` clause? Should comparisons now throw an error if I do e.g. `SELECT enum='' FROM tbl`

Reply:
Basically, we should always default to a try cast, unless is explicit?

Reply:
Perhaps the `AddCastToType` could take a `try_cast` parameter (default to false) that is set to true if any of the inputs is an enum?